proxmox/api/ticket_auth.go

```diff
@@ -29,15 +31,91 @@ type ticketAuthenticator struct {
 	mu sync.Mutex
 }
 
+// custom bool to handle bools of: int, text, bool // ref: https://stackoverflow.com/questions/4278430/convert-string-to-integer-type-in-go
+func (bit *cBool) UnmarshalJSON(inBits []byte) error {
+	inString := strings.ToLower(string(inBits))
+
+	inInt, err := strconv.Atoi(inString)
+	if err != nil {
+		return fmt.Errorf("failed injesting bool: %s", inString)
+	}
+
+	switch {
+	case inInt > 0 || inString == "true":
+		*bit = true
+	case inInt == 0 || inString == "false":
+		*bit = false
+	default:
+		return fmt.Errorf("cannot unmarshal boolean: %s", inString)
+	}
+
+	return nil
+}
+
 // NewTicketAuthenticator returns a new ticket authenticator.
+// precedence:  AuthPayload > AuthTicket & CSRFPreventionToken > OTP > user+pass // aka:  pre-auth > OTP > user+pass
 func NewTicketAuthenticator(conn *Connection, creds *Credentials) (Authenticator, error) {
+	if creds.AuthPayload != "" {
```

Comment:
@bpg im curious if this logic block would be better in `proxmox/api/client.go` (L 131 ish )

Reply:
I have a better pattern in my mind, all this conditional logic is getting a bit out of hands.
Let's decide on the payload vs ticket+csrf fist, and then I'll help you with refactoring of all of this to a more manageable structure.